packages/gqty/src/Helpers/track.ts

```diff
@@ -0,0 +1,175 @@
+import { GQtyError } from '../Error';
+import { debounce } from '../Utils/debounce';
+import { isAnySelectionIncluded } from '../Utils/selectionsInclude';
+
+import type { Selection } from '../Selection';
+import type { InnerClientState, SubscriptionsClient } from '../Client';
+
+export type TrackCallType = 'initial' | 'cache_change';
+
+export interface TrackCallInfo {
+  type: TrackCallType;
+}
+
+export interface TrackOptions {
+  onError?: ((err: GQtyError) => void) | undefined;
+
+  /**
+   * Refetch on initial call
+   */
+  refetch?: boolean;
+}
+
+export interface Track {
+  <Data>(callback: (info: TrackCallInfo) => Data, options?: TrackOptions): {
```

Comment:
What do you think about `callback: (info: TrackCallInfo) => CleanUpFunction`? And then the last cleanupFunction gets called on `stop()`?

Any special reason to return `Data` or some usage in mind? This cleanupFn is just to align with other popular API's like RxJS, React's `useEffect`

Reply:
the CleanUp function is the "stop" function, to use in useEffect is simply

```ts
const { stop } = track(() => {
  // ...
});

return stop
```

I'm not returning only the cleanup because that limits too much, and the name "stop" could be any other name, open to suggestions.

The generic in the callback was in place so that we can re-use the type and data returned from the callback, I will add another object returned by `track`, I will call it "data" with inside a field "current" with the latest data returned from the callback

Reply:
Alright! I mean `cleanupFn` as a return from the `callback`, not from track. 

But this answered: 
> I will add another object returned by track, I will call it "data" with inside a field "current" with the latest data returned from the callback

So you have intentions with this callback to get the lastest data. Alright :)

Reply:
and the date object reference is already there https://github.com/gqty-dev/gqty/pull/43/commits/084bba393bd2d3275e17658b024cf38ef6d41188 if you feel like this helper is fine as it is right now, I can just merge it